fix maven source compilation problem

[INFO] Compiling 4 source files to /home/travis/build/visze/simdrom/simdrom-cli/target/classes
/home/travis/build/visze/simdrom/simdrom-cli/src/main/java/de/charite/compbio/simdrom/cli/SIMdromSetting.java:292: error: multi-catch statement is not supported in -source 1.5
		} catch (NotAllowedCombinationOfOptionsException | MissingOptionsException e) {

--- a/pom.xml
+++ b/pom.xml
@@ -1,15 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <groupId>de.charite.compbio</groupId>
-  <artifactId>simdrom</artifactId>
-  <packaging>pom</packaging>
-  <version>0.0.1-SNAPSHOT</version>
-  <name>simdrom</name>
-  <properties>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>de.charite.compbio</groupId>
+	<artifactId>simdrom</artifactId>
+	<packaging>pom</packaging>
+	<version>0.0.1</version>
+	<name>simdrom</name>
+	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 	</properties>
- <dependencies>
+	<dependencies>
 		<dependency>
 			<groupId>com.google.guava</groupId>
 			<artifactId>guava</artifactId>
@@ -22,8 +23,52 @@
 			<scope>test</scope>
 		</dependency>
 	</dependencies>
- <modules>
- 	<module>simdrom-cli</module>
-  <module>simdrom-core</module>
-  </modules>
+	<modules>
+		<module>simdrom-cli</module>
+		<module>simdrom-core</module>
+	</modules>
+
+	<build>
+		<plugins>
+			<!-- JUnit testing -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-surefire-plugin</artifactId>
+				<version>2.18.1</version>
+				<configuration>
+					<skipTests>false</skipTests>
+					<forkCount>3</forkCount>
+					<reuseForks>true</reuseForks>
+					<argLine>${argLine} -Xms512m -Xmx1024m</argLine>
+					<systemPropertyVariables>
+						<user.language>en</user.language>
+						<user.region>GB</user.region>
+					</systemPropertyVariables>
+				</configuration>
+			</plugin>
+
+			<!-- Javadoc generation. -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-javadoc-plugin</artifactId>
+				<version>2.10.3</version>
+				<!-- Uncomment this to get the Javadoc to show all members <configuration> 
+					<show>private</show> </configuration> -->
+			</plugin>
+
+			<!-- Compilation -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.3</version>
+				<configuration>
+					<source>1.7</source>
+					<target>1.7</target>
+					<compilerArgument>-Xlint:all</compilerArgument>
+					<showWarnings>true</showWarnings>
+					<showDeprecation>true</showDeprecation>
+				</configuration>
+			</plugin>
+		</plugins>
+	</build>
 </project>

--- a/simdrom-cli/pom.xml
+++ b/simdrom-cli/pom.xml
@@ -1,20 +1,21 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <parent>
-    <groupId>de.charite.compbio</groupId>
-    <artifactId>simdrom</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
-  </parent>
-  <artifactId>simdrom-cli</artifactId>
-  <packaging>jar</packaging>
-  <name>simdrom-cli</name>
-  
-  
-  <dependencies>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>de.charite.compbio</groupId>
+		<artifactId>simdrom</artifactId>
+		<version>0.0.1</version>
+	</parent>
+	<artifactId>simdrom-cli</artifactId>
+	<packaging>jar</packaging>
+	<name>simdrom-cli</name>
+
+
+	<dependencies>
 		<dependency>
 			<groupId>de.charite.compbio</groupId>
 			<artifactId>simdrom-core</artifactId>
-			<version>0.0.1-SNAPSHOT</version>
+			<version>0.0.1</version>
 		</dependency>
 		<dependency>
 			<groupId>commons-cli</groupId>
@@ -26,65 +27,54 @@
 			<artifactId>commons-lang3</artifactId>
 			<version>3.4</version>
 		</dependency>
-  </dependencies>
-  
-  
-  <build>
-    <plugins>
+	</dependencies>
+
+
+	<build>
+		<plugins>
+			<!-- Make an executable jar and specify the main class and classpath -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jar-plugin</artifactId>
+				<version>2.6</version>
+				<configuration>
+					<archive>
+						<manifest>
+							<addClasspath>true</addClasspath>
+							<classpathPrefix>lib/</classpathPrefix>
+							<!-- set useUniqueVersions=false in order that the classpath has the 
+								SNAPSHOT instead of the build number prefixed to the dependency -->
+							<useUniqueVersions>false</useUniqueVersions>
+							<mainClass>de.charite.compbio.simdrom.Main</mainClass>
+						</manifest>
+					</archive>
+					<compilerArgument>-Xlint:all</compilerArgument>
+					<showWarnings>true</showWarnings>
+					<showDeprecation>true</showDeprecation>
+				</configuration>
+			</plugin>
+			     <!-- More JAR building -->
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.1</version>
-        <configuration>
-          <compilerId>javac-with-errorprone</compilerId>
-          <forceJavacCompilerUse>true</forceJavacCompilerUse>
-        </configuration>
-        <dependencies>
-          <dependency>
-            <groupId>org.codehaus.plexus</groupId>
-            <artifactId>plexus-compiler-javac-errorprone</artifactId>
-            <version>2.5</version>
-          </dependency>
-        </dependencies>
-      </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-surefire-plugin</artifactId>
-        <version>2.18.1</version>
-      </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-jar-plugin</artifactId>
-        <version>2.4</version>
-        <configuration>
-          <archive>
-            <manifest>
-              <addClasspath>true</addClasspath>
-              <mainClass>de.charite.compbio.simdrom.Main</mainClass>
-              <classpathPrefix>lib/</classpathPrefix>
-            </manifest>
-          </archive>
-        </configuration>
-      </plugin>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-dependency-plugin</artifactId>
-        <version>2.5.1</version>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>2.2</version>
         <executions>
           <execution>
-            <id>copy-dependencies</id>
             <phase>package</phase>
             <goals>
-              <goal>copy-dependencies</goal>
+              <goal>shade</goal>
             </goals>
             <configuration>
-              <outputDirectory>
-                ${project.build.directory}/lib/
-              </outputDirectory>
+              <transformers>
+                <transformer
+                  implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                  <mainClass>de.charite.compbio.simdrom.Main</mainClass>
+                </transformer>
+              </transformers>
             </configuration>
           </execution>
         </executions>
       </plugin>
-    </plugins>
-  </build>
+		</plugins>
+	</build>
 </project>

--- a/simdrom-core/pom.xml
+++ b/simdrom-core/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>de.charite.compbio</groupId>
     <artifactId>simdrom</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
+    <version>0.0.1</version>
   </parent>
   <artifactId>simdrom-core</artifactId>
   <name>simdrom-core</name>
@@ -29,4 +29,30 @@
   		<classifier>site</classifier>
   	</dependency>
   </dependencies>
+  
+  <build>
+    <plugins>
+      <!-- Specify the dependencies to copy -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <version>2.10</version>
+        <executions>
+          <execution>
+            <id>copy-dependencies</id>
+            <phase>package</phase>
+            <goals>
+              <goal>copy-dependencies</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.build.directory}/lib</outputDirectory>
+              <overWriteReleases>false</overWriteReleases>
+              <overWriteSnapshots>false</overWriteSnapshots>
+              <overWriteIfNewer>true</overWriteIfNewer>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+    </plugins>
+  </build>
 </project>

